verl/workers/actor/dp_actor.py

```diff
@@ -215,22 +224,31 @@ def compute_log_prob(self, data: DataProto) -> torch.Tensor:
             micro_batches = batch.split(micro_batch_size)
 
         log_probs_lst = []
+        metrics = {}
         for micro_batch in micro_batches:
             if isinstance(micro_batch, DataProto):
                 micro_batch = {**micro_batch.batch, **micro_batch.non_tensor_batch}
 
+            response_mask = micro_batch['attention_mask'][:, -micro_batch['responses'].size(-1):]
             with torch.no_grad():
-                _, log_probs = self._forward_micro_batch(micro_batch, temperature=temperature)
+                entropy, log_probs = self._forward_micro_batch(micro_batch,
+                                                               temperature=temperature,
+                                                               calculate_entropy=calculate_entropy)
             log_probs_lst.append(log_probs)
-        log_probs = torch.concat(log_probs_lst, dim=0)
+            if calculate_entropy:
```

Comment:
Can we return full entropy to the driver and the driver compute entropy? We may need per-token entropy for analysis